packages/core/src/styling/theme.ts

```diff
@@ -71,49 +71,85 @@ export type OldTheme = {
   breakpoints: Record<string, { value: number; order: number }>
 }
+
 export type Theme = {
+  default?: true
+  defaultDark?: true
+  defaultLight?: true
+  propertyDefinitions?: Record<CustomPropertyName, CustomPropertyDefinition>
+
   scheme?: 'dark' | 'light'
-  color: StyleTokenGroup[]
+  color?: StyleTokenGroup[]
   fonts: FontFamily[]
-  'font-size': StyleTokenGroup[]
-  'font-weight': StyleTokenGroup[]
-  spacing: StyleTokenGroup[]
-  'border-radius': StyleTokenGroup[]
-  shadow: StyleTokenGroup[]
-  'z-index': StyleTokenGroup[]
-  categories?: Record<string, CustomCategory>
-}
-
-export type CustomCategory = {
-  propertyDefinitions: Record<CustomPropertyName, CustomPropertyDefinition>
+  'font-size'?: StyleTokenGroup[]
+  'font-weight'?: StyleTokenGroup[]
+  spacing?: StyleTokenGroup[]
+  'border-radius'?: StyleTokenGroup[]
+  shadow?: StyleTokenGroup[]
+  'z-index'?: StyleTokenGroup[]
 }
 
 export type CustomPropertyDefinition = {
   syntax: CssSyntaxNode
   inherits: boolean
-  initialValue: string
+  initialValue: string | null // Required by CSS specs for default-theme, but we can do a fallback so null is allowed
   description: string
 }
 
-export const getThemeCss = (theme: Theme | OldTheme, options: ThemeOptions) => {
-  if ('breakpoints' in theme) {
-    return getOldThemeCss(theme)
-  }
+export const getThemeCss = (
+  theme: Record<string, OldTheme | Theme>,
+  options: ThemeOptions,
+) => {
+  const [themesV1, themesV2] = Object.entries(theme).reduce(
+    ([legacy, modern], [key, value]) => {
+      if ('breakpoints' in value) {
+        legacy[key] = value
+      } else {
+        modern[key] = value
+      }
+      return [legacy, modern]
+    },
+    [{}, {}] as [Record<string, OldTheme>, Record<string, Theme>],
+  )
 
-  return `${Object.values(theme.categories ?? {})
-    .map((category) =>
-      Object.entries(category.propertyDefinitions)
-        .map(([propertyName, property]) =>
-          renderSyntaxDefinition(propertyName as CustomPropertyName, property),
-        )
-        .join('\n'),
+  const defaultTheme =
+    Object.values(themesV2).find((t) => t.default) ?? Object.values(themesV2)[0]
+  const defaultDarkTheme = Object.values(themesV2).find((t) => t.defaultDark)
+  const defaultLightTheme = Object.values(themesV2).find((t) => t.defaultLight)
+
+  return `
+  ${Object.values(themesV1)
+    .map((t) => getOldThemeCss(t))
+    .join('\n')}
+
+  ${Object.entries(defaultTheme.propertyDefinitions ?? {})
+    .map(([propertyName, property]) =>
+      renderSyntaxDefinition(
+        propertyName as CustomPropertyName,
+        property,
+        defaultTheme,
+      ),
     )
     .join('\n')}
+
+  /* Render default theme */
+  ${renderTheme(':host, :root', defaultTheme)}
+
+  ${renderTheme(':host, :root', defaultDarkTheme, '@media (prefers-color-scheme: dark)')}
+  ${renderTheme(':host, :root', defaultLightTheme, '@media (prefers-color-scheme: light)')}
```

Comment:
Consider creating a function that renders multiple themes. Then, if no dark/light theme exist, that function could make sure we don't add unnecessary newlines.